packages/tools/client-debugger/client-debugger-view/src/components/data-object-views/SharedMapView.tsx

```diff
@@ -44,28 +48,43 @@ export function SharedMapView(props: SharedMapViewProps): React.ReactElement {
 		};
 	}, [sharedMap, setEntries]);
 
+	const toggleCollapse = (key: string): void => {
+		setCollapsedEntries({
+			...collapsedEntries,
+			[key]: !collapsedEntries[key],
+		});
+	};
+
 	return (
-		<table style={{ borderCollapse: "collapse", width: "100%" }}>
-			<thead>
-				<tr>
-					<th>Key</th>
-					<th>Value</th>
-				</tr>
-			</thead>
-			<tbody style={{ borderCollapse: "collapse" }}>
-				{entries.map(([key, value]) => (
-					<tr key={key} style={{ borderCollapse: "collapse", border: "thin solid" }}>
-						<td
-							data-label="Key"
-							style={{ borderCollapse: "collapse", border: "thin solid" }}
-						>
-							{key}
-						</td>
-						<td data-label="Value">{getTableValue(value, renderChild)}</td>
-					</tr>
-				))}
-			</tbody>
-		</table>
+		<div>
+			{entries.map(([key, value]) => (
```

Comment:
Can we extract this drop-down tree-view pattern into its own component? That way all of our views could just call that and get consistent functionality.